appscale/tools/appscale_tools.py

```diff
@@ -505,14 +505,26 @@ def run_instances(cls, options):
       AppScaleLogger.log("\nPlease wait for AppScale to prepare your machines "
                          "for use. This can take few minutes.")
 
+      head_node.public_ip = public_ips[0]
```

Comment:
At this time we have no good way of knowing the position of the head node in node_layout.nodes.
Since the Azure agent only returns a public IP for the head node, we are updating that first with a different accessor (head_node) and then we update the rest of the list with just private IPs.

The real solution would be to modify the NodeLayout parser to deterministically put the head node(s)/loadbalancer(s) first in the list. That was too much work at the time.